### PR TITLE
[arci-gamepad-keyboard] Add arci-gamepad-keyboard

### PR DESCRIPTION
**Pull request by taiki-e**
This adds a new crate `arci-gamepad-keyboard`, and `gamepad = "Keyboard"` config of `openrr_apps_robot_teleop`.

## Examples

```bash
# first, open `./openrr-apps/config/sample_teleop_config_urdf_viz.toml` and set `gamepad = "Keyboard"` manually

# second, launch urdf-viz
urdf-viz ./openrr-planner/sample.urdf &

# next, launch openrr_apps_robot_teleop
cargo run -p openrr-apps --bin openrr_apps_robot_teleop -- --config-path ./openrr-apps/config/sample_teleop_config_urdf_viz.toml
```

## Key mappings

```text
LeftStick:
   q    w    e
   a    s    d
   z    x    c

RightStick:
   u    i    o
   j    k    l
   m    ,    .

5 : ^ (DPadUp)
r : < (DPadLeft)
t : > (DPadRight)
f : v (DPadDown)

y : △ (North)
g : □ (West)
h : ○ (East)
b : x (South)

1 : L1 (LeftTrigger)
2 : L2 (LeftTrigger2)
3 : L3 (LeftThumb)

8 : R1 (RightTrigger)
9 : R2 (RightTrigger2)
0 : R3 (RightThumb)

6 : Select
7 : Start
```

### CODE REVIEW

**Comment by OTL** [comment]
Isn't the above description wrong? (copy paste?)

```test
LeftStick:
   u    i    o
   j    k    l
   m    ,    .

RightStick:
   u    i    o
   j    k    l
   m    ,    .
```

File: arci-gamepad-keyboard/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+cargo run -p openrr-apps --bin openrr_apps_robot_teleop_keyboard -- --config-path ./openrr-apps/config/sample_teleop_config_urdf_viz.toml
+```
+
+## Key mappings
+
+```text
+LeftStick:
+   u    i    o
+   j    k    l
+   m    ,    .
```
Comment (OTL) [inline arci-gamepad-keyboard/README.md:22]: ?

File: arci-gamepad-keyboard/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[package]
+name = "arci-gamepad-keyboard"
+version = "0.0.5"
+authors = ["Taiki Endo <te316e89@gmail.com>"]
```
Comment (OTL) [inline arci-gamepad-keyboard/Cargo.toml:4]: Please update mail address

File: arci-gamepad-keyboard/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+//! Currently, this crate only supports Unix-like operating systems.
+//!
+//! # Key mappings
+//!
+//! ```text
+//! LeftStick:
+//!    u    i    o
+//!    j    k    l
+//!    m    ,    .
+//!
```
Comment (OTL) [inline arci-gamepad-keyboard/src/lib.rs:12]: ditto

File: openrr-apps/src/bin/robot_teleop_keyboard.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
+#![cfg_attr(windows, allow(dead_code, unused_imports))]
```
Comment (OTL) [inline openrr-apps/src/bin/robot_teleop_keyboard.rs:1]: I don't want to create this.
Is it difficult to modify robot_teleop.rs to support teleop-keyboard?
(That's the concept of arci.)

Comment (taiki-e) [reply openrr-apps/src/bin/robot_teleop_keyboard.rs:1]: I suppose I could add a property like `gamepad = "Gilrs" | "Keyboard"` to the teleop config.

```rust
pub enum GamepadKind {
    Gilrs,
    Keyboard,
}
```

...but I'm not sure that would be easy to use without #295.

File: arci-gamepad-keyboard/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+            sender,
+            key_state,
+            button_map: button_map(),
+        }
+    }
+
+    fn right_stick(&mut self, sign_x: i8, sign_y: i8) {
+        self.send(GamepadEvent::AxisChanged(
+            Axis::RightStickX,
+            sign_x as f64 * 0.3,
```
Comment (OTL) [inline arci-gamepad-keyboard/src/lib.rs:95]: I want to define a magic number as a const value, like

```rust
const XXX: f64 = 0.3;
```

File: openrr-apps/src/bin/robot_teleop_keyboard.rs (hunk 4 above)
Comment (OTL) [reply openrr-apps/src/bin/robot_teleop_keyboard.rs:1]: Even if it's not easy to use, I want you to do that.

And I noticed that Windows is not supported, right?

Comment (taiki-e) [reply openrr-apps/src/bin/robot_teleop_keyboard.rs:1]: > Even if it's not easy to use, I want you to do that.

Ok, will do.

> And I noticed that Windows is not supported, right?

Yeah, termios doesn't support windows.

Comment (taiki-e) [reply openrr-apps/src/bin/robot_teleop_keyboard.rs]: done

**Review by OTL — APPROVED** [review]

**Pull request merged**